bpf/process/bpf_execve_event.c

```diff
@@ -191,11 +192,16 @@ read_exe(struct task_struct *task, struct heap_exe *exe)
 	// buffer used by d_path_local can contain up to MAX_BUF_LEN i.e. 4096 we
 	// only keep the first 255 chars for our needs (we sacrifice one char to the
 	// verifier for the > 0 check)
-	if (exe->len > 255)
+	if (exe->len > 255) {
+		offset = exe->len - 255;
 		exe->len = 255;
```

Comment:
a note not on your PR but general improvement, we could replace that with `BINARY_PATH_MAX_LEN - 1` here.